TestMatchBinariesFollowChildrenUpdate flake on rhel8

### What happened? How can we reproduce this?

looks like there's a flake in TestMatchBinariesFollowChildrenUpdate on rhel8,
which occasionally fails with:

```
level=info msg="BTF discovery: default kernel btf file found" btf-file=/sys/kernel/btf/vmlinux
=== RUN   TestMatchBinariesFollowChildrenUpdate
    logcapture.go:24: time=2025-12-29T15:16:29.956Z level=INFO msg="Exit probe on acct_process"
    logcapture.go:24: time=2025-12-29T15:16:29.959Z level=INFO msg="Set execve_map entries 32768" size=28M
    logcapture.go:24: time=2025-12-29T15:16:29.976Z level=INFO msg="BTF file: using metadata file" metadata=/sys/kernel/btf/vmlinux
    logcapture.go:24: time=2025-12-29T15:16:29.976Z level=INFO msg="Loading sensor" name=__base__
    logcapture.go:24: time=2025-12-29T15:16:29.976Z level=INFO msg="Loading kernel version 4.18.0"
    logcapture.go:24: time=2025-12-29T15:16:30.130Z level=ERROR msg="detect link pin" error="perf event ioctl pin: not supported"
    logcapture.go:24: time=2025-12-29T15:16:30.348Z level=INFO msg="Loaded sensor successfully" sensor=__base__
    logcapture.go:24: time=2025-12-29T15:16:30.366Z level=INFO msg="Read ProcFS /proc appended 139/197 entries"
    logcapture.go:24: time=2025-12-29T15:16:30.369Z level=INFO msg="BTF file: using metadata file" metadata=/sys/kernel/btf/vmlinux
    logcapture.go:24: time=2025-12-29T15:16:30.370Z level=INFO msg="Loading sensor" name=test-sensor-1
    logcapture.go:24: time=2025-12-29T15:16:30.370Z level=INFO msg="Loading kernel version 4.18.0"
    logcapture.go:24: time=2025-12-29T15:16:30.380Z level=INFO msg="Loaded sensor successfully" sensor=test-sensor-1
    logcapture.go:24: time=2025-12-29T15:16:30.481Z level=INFO msg="Added kprobe" return=false function=__x64_sys_prctl override=false
    logcapture.go:24: time=2025-12-29T15:16:30.482Z level=INFO msg="Added generic kprobe sensor: /home/runner/work/tetragon/tetragon/go/src/github.com/cilium/tetragon/bpf/objs/bpf_generic_kprobe_v53.o -> __x64_sys_prctl" override=false
    logcapture.go:24: time=2025-12-29T15:16:30.482Z level=INFO msg="BTF file: using metadata file" metadata=/sys/kernel/btf/vmlinux
    logcapture.go:24: time=2025-12-29T15:16:30.482Z level=INFO msg="Loading sensor" name=generic_kprobe
    logcapture.go:24: time=2025-12-29T15:16:30.482Z level=INFO msg="Loading kernel version 4.18.0"
    logcapture.go:24: time=2025-12-29T15:16:31.415Z level=INFO msg="Loaded generic kprobe program: /home/runner/work/tetragon/tetragon/go/src/github.com/cilium/tetragon/bpf/objs/bpf_generic_kprobe_v53.o -> __x64_sys_prctl"
    logcapture.go:24: time=2025-12-29T15:16:31.415Z level=INFO msg="Loaded sensor successfully" sensor=generic_kprobe
    matchbinaries_follow_children_test.go:302:
                Error Trace:    /home/runner/work/tetragon/tetragon/go/src/github.com/cilium/tetragon/pkg/sensors/tracing/matchbinaries_follow_children_test.go:302
                Error:          Not equal:
                                expected: 0x1
                                actual  : 0x0
                Test:           TestMatchBinariesFollowChildrenUpdate
                Messages:       Binary.MBSet_1
    logcapture.go:24: time=2025-12-29T15:16:31.423Z level=INFO msg="Unloading sensor test-sensor-1"
    logcapture.go:24: time=2025-12-29T15:16:31.427Z level=INFO msg="Sensor unloaded" sensor=test-sensor-1 maps-error=[]
    logcapture.go:24: time=2025-12-29T15:16:31.427Z level=INFO msg="Unloading sensor __base__"
    logcapture.go:24: time=2025-12-29T15:16:31.470Z level=INFO msg="Sensor unloaded" sensor=__base__ maps-error=[]
    base.go:202: cleanup: unloading base sensor
    logcapture.go:24: time=2025-12-29T15:16:31.471Z level=INFO msg="Unloading sensor __base__"
--- FAIL: TestMatchBinariesFollowChildrenUpdate (1.58s)
FAIL

```

### Tetragon Version

latest v1.7.0

### Kernel Version

rhel8 v4.18

### Kubernetes Version

_No response_

### Bugtool

_No response_

### Relevant log output

```shell

```

### Anything else?

_No response_

--- a/pkg/sensors/tracing/matchbinaries_follow_children_test.go
+++ b/pkg/sensors/tracing/matchbinaries_follow_children_test.go
@@ -298,8 +298,11 @@ func TestMatchBinariesFollowChildrenUpdate(t *testing.T) {
 		found = 0
 		iter := hash.Iterate()
 		for iter.Next(&key, &val) {
-			if unix.ByteSliceToString(val.Binary.Path[:]) == forks {
-				require.Equal(t, uint64(1), val.Binary.MBSet, "Binary.MBSet_1")
+			// We need to ccount execve_map records with:
+			// - forks binary path
+			// - val.Binary.MBSet == 1
+			if unix.ByteSliceToString(val.Binary.Path[:]) == forks &&
+				val.Binary.MBSet == 1 {
 				found++
 			}
 		}

